p4-fusion/git_api.h

```diff
@@ -17,30 +18,57 @@
 
 struct git_repository;
 
+class BlobWriter
+{
+private:
+	git_repository* repo;
+	git_writestream* writer;
+	bool begun;
+	bool finalized;
+
+public:
+	BlobWriter() = delete;
+	BlobWriter(git_repository* repo);
+
+	// Write creates a new ODB entry on the first call and continuous calls keep
+	// writing more data to it.
+	void Write(const char* contents, int length);
+	// Close MUST be called at the end of the writing process to finalize the ODB entry
+	// and to move it into it's proper place on disk.
+	std::string Close();
+};
+
 class GitAPI
 {
 	git_repository* m_Repo = nullptr;
 	git_oid m_FirstCommitOid;
+	std::string repoPath;
 	int timezoneMinutes;
+	std::map<std::string, git_index*> lastBranchTree;

```

Comment:
Use `std::unordered_map` as that's a hash table (albeit a bad one). `std::map` is implemented as a red-black tree.